### ISSUE TITLE
[java] CompareObjectsWithEquals: FP with array.length

### ISSUE THREAD

**Issue by djh82** [pmd/pmd#2976]
Since: 6.30.0

Rule: [CompareObjectsWithEquals](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#compareobjectswithequals)

Codebase passes with previous PMD version, just upgraded to 6.30.0 and are now seeing assumed false positives with CompareObjectsWithEquals.

Code is as follows:

```java
public static byte[] myArrayFunc(byte[] a1, byte[] a2) {
    if (a1.length != a2.length) {
        throw new IllegalArgumentException();
    }

    ....
}
```

PMD now fails on the `if (a1.length != a2.length)` line; they are both primitive ints, and therefore do not have an equals method.

Possibly introduced by #2934

**Comment by catellie** [comment]
For the record: I see exactly this issue with strings when trying to upgrade to 6.30.0.

**Comment by zman0900** [comment]
Also happening when comparing the results of two method calls that return long.  For example:

```java
class Crap {
    private final long val;

    public long getVal() { return val; }
}

if (c1.getVal() != c2.getVal()) {  // <-- here
    // whatever
}
```

**Comment by armiol** [comment]
Guys,

With Java 11's `var` I can still reproduce the same issue. Here is the simplified sample:

```java
    @SuppressWarnings("UseOfSystemOutOrSystemErr")
    public static void main(String[] args) {
        var first = "first";
        var second = "second";
        var firstBytes = first.getBytes(UTF_8);
        var secondBytes = second.getBytes(UTF_8);
        if(firstBytes.length != secondBytes.length) {
            System.out.println("PMD should have failed at the line above.");
        }
    }
```

Which gives

```
 CompareObjectsWithEquals:       Use equals() to compare object references.
 UseEqualsToCompareStrings:      Use equals() to compare strings instead of '==' or '!='
```

PMD versions: 6.41.0
OpenJDK 12.0.1

Interestingly, the version without `var`s passes the inspections just fine:

```java
    @SuppressWarnings({"UseOfSystemOutOrSystemErr", "RedundantExplicitVariableType"})
    public static void main(String[] args) {
        String first = "first";
        String second = "second";
        byte[] firstBytes = first.getBytes(UTF_8);
        byte[] secondBytes = second.getBytes(UTF_8);
        if(firstBytes.length != secondBytes.length) {
            System.out.println("PMD should have failed at the line above.");
        }
    }
```

**Comment by adangel** [comment]
@armiol Please create a new issue for that. This original issue was not about using `var`, so your case is different. Thanks!

**Comment by sroeper** [comment]
I still have the issue with size of Lists:
```java
List<String> list1 = new ArrayList<>();
List<String> list2 = new ArrayList<>();

if (list1.size() != list2.size()) {   // <-- reports CompareObjectsWithEquals
  // do something
}
```
